content/en/smart-contracts/filecoin-evm-runtime/differences-with-ethereum/index.md

```diff
@@ -13,32 +13,44 @@ weight: 100
 
-Pug gluten-free scenester mustache sartorial hoodie. Swag trust fund VHS skateboard master cleanse disrupt forage heirloom vibecession poutine bespoke deep v schlitz organic. DIY green juice pok pok pinterest DSA tilde ethical. Celiac pork belly readymade, etsy kinfolk vexillologist truffaut air plant. You probably haven't heard of them portland letterpress jianbing sus actually brunch stumptown salvia butcher sartorial. Squid taiyaki activated charcoal bushwick umami viral.
+1. EVM and Filecoin gas are different units of measurement and are not 1:1. Purely based on chain throughput (gas/second), the ratio of Ethereum gas to Filecoin gas is about 1:444. Expect Filecoin gas numbers to look _much_ larger than those in Ethereum.
+1. Because Filecoin charges Filecoin gas for executing the Filecoin EVM runtime interpreter:
+    1. Some instructions may be more expensive and/or cheaper in Filecoin EVM runtime than they are in the EVM.
+    1. EVM instruction costs can depend on the exact Filecoin EVM runtime code-paths taken, and caching.
 
-### Heirloom
+{{< alert >}}
+⚠️ Filecoin gas costs are not set in stone and should never be hard-coded. Future network upgrades will break any smart contracts that depend on gas costs not changing.
+{{< /alert >}}
 
-Banh mi mixtape swag lumbersexual jean shorts, jianbing PBR&B pok pok lomo meditation hammock actually fashion axe squid gochujang. Squid poke shabby chic church-key mlkshk schlitz. Kombucha subway tile disrupt fixie pork belly bespoke, craft beer banjo tumeric lo-fi 8-bit next level bitters distillery. Squid XOXO yuccie authentic. Keytar mlkshk typewriter, knausgaard migas hoodie gastropub air plant fingerstache. Heirloom salvia 3 wolf moon shaman.
+## Gas stipend
 
-Iceland next level literally, butcher pok pok gentrify readymade shaman. Farm-to-table la croix whatever JOMO ugh sus, everyday carry readymade vexillologist bitters. +1 blog intelligentsia hashtag umami, celiac vice photo booth. Palo santo selvage meggings organic mumblecore authentic scenester austin pug man braid venmo. Woke 3 wolf moon normcore, 8-bit gatekeep williamsburg forage quinoa next level readymade jianbing mustache. Trust fund swag godard tumblr chicharrones mlkshk vaporware.
+Solidity calls to `address.transfer` and `address.send` grant a fixed gas stipend of 2300 gas to the called contract. Unfortunately, this is EVM gas, not Filecoin gas. To compensate, Filecoin EVM runtime automatically sets the gas limit to 10 million Filecoin gas if it looks like a `transfer` or `send`.
```

Comment:
Suggest a simplification and a more straightforward conclusion:

> Solidity calls to `address.transfer` and `address.send` grant a fixed gas stipend of 2300 Ethereum gas to the called contract. The Filecoin EVM runtime automatically detects such calls, and sets the gas limit to 10 million Filecoin gas. This is a relatively more generous limit than Ethereum's, but it's future-proof. You should expect the callee to be able to carry out more work than in Ethereum.

Reply:
Resolved in [a92ed3d](https://github.com/filecoin-project/filecoin-docs/pull/1694/commits/a92ed3d39ff0db9643c4aaf765f47ed7258f6992)